commands/deploy_command.go

```diff
@@ -437,7 +437,27 @@ func (c *DeployCommand) doUploadFromUrl(encodedFileUrl string, mtaClient mtaclie
 		}
 		file = jobResult.File
 		if len(jobResult.Error) != 0 {
-			ui.Failed("Async upload job failed: %s", jobResult.Error)
+			var fileUrlWithoutCredentials string
+		
+			decodedBytes, err := base64.URLEncoding.DecodeString(encodedFileUrl)
+			if err != nil {
+				fmt.Println("Error decoding:", err)
+				fileUrlWithoutCredentials = "an unknown url" 
+			} else {
+				decodedURL := string(decodedBytes)
+		
+				parsedURL, err := url.Parse(decodedURL)
+				if err != nil {
+					fmt.Println("Error parsing URL:", err)
+					fileUrlWithoutCredentials = "an unknown url" 
+				} else {
+					parsedURL.User = nil
+					fileUrlWithoutCredentials = parsedURL.String() 
+				}
+			}
+		
+			ui.Failed("Async upload job failed: Credentials to %s are wrong. Make sure that they are correct.", fileUrlWithoutCredentials)
```

Comment:
Why do you print the URL? Is it really needed?
If the async job fails it is not always that the credentials are wrong, might be a network or some other issue. I'd suggest to rely solely on the backend for the error message.